overlord/snapstate/snapmgr.go

```diff
@@ -183,6 +184,40 @@ func (snapsup *SnapSetup) MountFile() string {
 	return snap.MountFileInDir(blobDir, snapsup.InstanceName(), snapsup.Revision())
 }
 
+// ComponentSetup holds the necessary component details to perform
+// most component related tasks.
+type ComponentSetup struct {
+	// CompSideInfo for metadata not coming from the component
+	CompSideInfo *snap.ComponentSideInfo `json:"comp-side-info,omitempty"`
+	// CompPath is the path to the file
+	CompPath string `json:"comp-path,omitempty"`
+
+	// Following fields are information related to the owner snap
+	// that we need in the tasks.
+
+	// SnapInstanceKey is set by the user during installation and
+	// differs for each instance of component owner snap.
+	SnapInstanceKey string `json:"instance-key,omitempty"`
+	// SnapType is the type of the owner snap
+	SnapType snap.Type `json:"snap-type,omitempty"`
+	// SnapRevision is the revision of the owner snap
+	SnapRevision snap.Revision `json:"snap-revision,omitempty"`
```

Comment:
as I said my preference would be to attach a partial SnapSetup instead, a bit like the ones we use for things like enable/disable etc

Reply:
Done now